src/wallet/rpcwallet.cpp

```diff
@@ -3007,6 +3011,112 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+    if (fHelp || params.size() < 1 || params.size() > 3)
+        throw runtime_error(
+            "z_getaddressforaccount account ( diversifier_index [\"pool\", ...] )\n"
+            "\nFor the given account number, return a UA in accordance with the remaining arguments."
+            "\nIf no diversifier index is given (or the string \"*\"), the next unused index will be selected."
+            "\nCompute a Unified Address corresponding to the diversifier index, containing addresses"
+            "\nfor the given pools (current default [\"transparent\",\"sapling\",\"orchard\"]).\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"account\": n,                    (numeric) the specified account number\n"
+            "  \"diversifier\": n,                (numeric) the index specified or chosen\n"
+            "  \"pools\": [\"pool\",...]\",          (json array of string) the pools (e.g. \"transparent\", \"orchard\") the UA contains\n"
+            "  \"unifiedaddress\"                 (string) The corresponding address\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("z_getaddressforaccount", "4")
+            + HelpExampleCli("z_getaddressforaccount", "4 1")
+            + HelpExampleCli("z_getaddressforaccount", "4 1 '[\"transparent\",\"sapling\",\"orchard\"]'")
+            + HelpExampleRpc("z_getaddressforaccount", "4")
+        );
+
+    if (!fExperimentalOrchardWallet) {
+        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: the Orchard wallet experimental extensions are disabled.");
+    }
+    int64_t account = params[0].get_int64();
+    if (account < 0 || account > 0x7fffffff-1) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid account number, must be 0 <= account <= (2^31)-2.");
+    }
+
+    std::vector<uint8_t> parsed_diversifier_index;
+    if (params.size() >= 2) {
+        if (params[1].getType() != UniValue::VNUM) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid diversifier index, must be an unsigned integer.");
+        }
+        auto parsed_diversifier_index_opt = ParseArbitraryInt(params[1].getValStr());
+        if (!parsed_diversifier_index_opt.has_value()) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "diversifier index must be a decimal integer.");
+        }
+        parsed_diversifier_index = parsed_diversifier_index_opt.value();
+        if (parsed_diversifier_index.size() > ZC_DIVERSIFIER_SIZE) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "diversifier index is too large.");
+        }
+    } else {
+        // TODO get next unused diversifier index from wallet
+    }
+    diversifier_t diversifier{};
+    // expand vector just to print, there is probably a better way
+    for (int i = 0; i < parsed_diversifier_index.size(); ++i) diversifier[i] = parsed_diversifier_index[i];
```

Comment:
`diversifier_t` is a typedef for `std::array<unsigned char, ZC_DIVERSIFIER_SIZE>`, so you should be able to write
```suggestion
    std::copy(parsed_diversifier_index.begin(), parsed_diversifier_index.end(), diversifier.begin());
```

Reply:
Hmm, I had written some debugging prints after this code, but it's been removed, so I don't think I need this at all. But later, when we fill in the functionality of this RPC, we'll need this `diversifier` variable, so I think I'll leave your suggested code but just comment it out.